src/library/autodj/autodjprocessor.cpp

```diff
@@ -1039,14 +1039,18 @@ void AutoDJProcessor::calculateTransition(DeckAttributes* pFromDeck,
     }
     double outroLength = outroEnd - outroStart;
 
+    double toDeckPosition = pToDeck->playPosition() * toTrackDuration;
+    pToDeck->fadeBeginPos = getOutroStartPosition(pToDeck);
```

Comment:
This may be inaccurate with the Full Intro + Outro mode, but I think it's good enough.

Reply:
Currently we only calculate one transition at a time, so getting the real next transition time would require a major refactoring. In the future that will be necessary to [calculate the remaining time in the playlist](https://bugs.launchpad.net/mixxx/+bug/1293980).

Reply:
pToDeck->fadeBeginPos is set again for the other transition types below. 
The issue here is that we have a common block for the first two transitions. This is not obvious, but fixing it will clutter the code in an other way. 

We do not calculate the real next transition time here. We just store a limiting position here which is used to decide if we need to re-cue the track or not. 

I will add a comment.